File: src/main/java/zaftnotameni/creatania/util/Text.java
package zaftnotameni.creatania.util;
import com.simibubi.create.foundation.utility.Lang;
import com.simibubi.create.foundation.utility.LangBuilder;
import net.minecraft.ChatFormatting;
import net.minecraft.network.chat.Component;
public class Text {
  public static LangBuilder colored(String text, ChatFormatting color) {
    return Lang.builder().add(Component.literal(text)).style(color);
  }
  public static LangBuilder muted(String text) { return colored(text, ChatFormatting.DARK_GRAY); }
  public static LangBuilder gray(String text) { return colored(text, ChatFormatting.GRAY); }
  public static LangBuilder green(String text) { return colored(text, ChatFormatting.DARK_GREEN); }
  public static LangBuilder aqua(String text) {
    return colored(text, ChatFormatting.AQUA);
  }
  public static LangBuilder red(String text) {
    return colored(text, ChatFormatting.RED);
  }
  public static LangBuilder purple(String text) { return colored(text, ChatFormatting.LIGHT_PURPLE); }
}

File: src/main/java/zaftnotameni/creatania/util/Text.kt
@file:Suppress("unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused",
  "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused", "unused"
)

package zaftnotameni.creatania.util

import com.simibubi.create.foundation.utility.Lang
import com.simibubi.create.foundation.utility.LangBuilder
import net.minecraft.ChatFormatting
import net.minecraft.network.chat.TextComponent

object Text {
  fun colored(text : String?, color : ChatFormatting?) : LangBuilder {
    return Lang.builder().add(TextComponent(text)).style(color)
  }

  fun muted(text : String?) : LangBuilder {
    return colored(text, ChatFormatting.DARK_GRAY)
  }

  fun gray(text : String?) : LangBuilder {
    return colored(text, ChatFormatting.GRAY)
  }

  @Suppress("unused") fun green(text : String?) : LangBuilder {
    return colored(text, ChatFormatting.DARK_GREEN)
  }

  fun aqua(text : String?) : LangBuilder {
    return colored(text, ChatFormatting.AQUA)
  }

  fun red(text : String?) : LangBuilder {
    return colored(text, ChatFormatting.RED)
  }

  fun purple(text : String?) : LangBuilder {
    return colored(text, ChatFormatting.LIGHT_PURPLE)
  }
}